pyiceberg/transforms.py

```diff
@@ -146,7 +146,13 @@ def can_transform(self, source: IcebergType) -> bool:
         return False
 
     @abstractmethod
-    def result_type(self, source: IcebergType) -> IcebergType: ...
+    def result_type(self, source: IcebergType) -> IcebergType:
+        """
+        Return the type of the result of the transform.
+
+        This type does not need to conform to the Iceberg spec, as long as it is converted to the correct type in the stored metadata.
```

Comment:
reading this as someone less familiar with the discussions in the other MR, this doesn't tell me much.  Would it be good to link that MR here so future folks have the context of the discussion?  Otherwise, might be good to go into more detail here.

Reply:
+1, I agree it's not capturing the whole context, let me see if I can come up with a suggestion.

Reply:
@kevinjqliu just following up on this. Do you have an alternative in mind? Otherwise, I will link the relevant PR here.